src/server/dfly_main.cc

```diff
@@ -96,6 +99,122 @@ namespace dfly {
+  auto const regex = std::regex(reg_match_expr);
+  std::smatch match;
+  if (std::regex_match(from, match, regex) && match.size() > 1) {
+    // the second entry is the match to the group that holds the version string
+    return match[1].str();
+  } else {
+    return std::string{kInvalidVersion};
+  }
+}
+
+std::string GetRemoteVersion(ProactorBase* proactor, SSL_CTX* ssl_context, const std::string host,
+                             std::string_view service, const std::string& resource) {
+  namespace bh = boost::beast::http;
+  using ResponseType = bh::response<bh::string_body>;
+
+  TlsClient http_client{proactor};
+
+  http_client.set_connect_timeout_ms(2000);
+  auto version = proactor->Await([&] {
+    if (auto ec = http_client.Connect(host, service, ssl_context); !ec) {
+      bh::request<bh::string_body> req{bh::verb::get, resource, 11 /*http 1.1*/};
+      req.set(bh::field::host, host.c_str());
+      ResponseType res;
+      if (auto ec = http_client.Send(req, &res); !ec) {
+        VLOG(1) << "successfully got response from HTTP GET for host " << host << ":" << service
+                << "/" << resource << " response code is " << res.result();
+
+        if (res.result() == bh::status::ok) {
+          const auto& body = res.body();
+          return GetVersionString(body);
+        }
+      } else {
+        LOG(WARNING) << "failed to process send HTTP GET to " << host << "@" << service
+                     << " at resource '" << resource << "'";
+      }
+    } else {
+      LOG(WARNING) << "failed to connect: " << ec.message();
+    }
+    return std::string{kInvalidVersion};
+  });
+
+  return version;
+}
+
+struct VersionMonitor {
+  fibers_ext::Fiber version_fiber_;
+  fibers_ext::Done monitor_ver_done_;
+  ProactorPool* proactor_pool_ = nullptr;
+
+  explicit VersionMonitor(ProactorPool* pp) : proactor_pool_{pp} {
+  }
+
+  void Run() {
+    version_fiber_ = proactor_pool_->GetNextProactor()->LaunchFiber([this] { RunTask(); });
+  }
+
+  void Shutdown() {
+    monitor_ver_done_.Notify();
+    if (version_fiber_.IsJoinable()) {
+      version_fiber_.Join();
+    }
+  }
+
+ private:
+  void RunTask();
+};
+
+void VersionMonitor::RunTask() {
+  const auto loop_sleep_time = std::chrono::hours(24);  // every 24 hours
+
+  const std::string host_name = "run-stage-43h632k3xq-ew.a.run.app";
+  const std::string_view port = "443";
+  const std::string resource = "/v1";
+  const std::string dev_version_name = "dev";
+
+  // Don't run in dev environment - i.e. where we don't build with
+  // real version number
+  if (kGitTag == dev_version_name) {
+    return;
+  }
+
+  SSL_CTX* context = TlsClient::CreateSslContext();
+  if (!context) {
+    LOG(WARNING) << "failed to create SSL context - cannot run version monitoring";
+    return;
+  }
+
+  auto CheckForNewerVersion = [&]() {
+    auto remote_version =
+        GetRemoteVersion(proactor_pool_->GetNextProactor(), context, host_name, port, resource);
```

Comment:
you do not need to run another call inside another proactor. You are already in the context of a fiber. running it in a different proactor fiber does not help you.

Reply:
remove the call to get next proactor and remove the pool